sleap/gui/commands.py

```diff
@@ -3470,13 +3481,230 @@ def do_action(context: CommandContext, params: dict):
+        video_fps = getattr(context.state["video"], "fps", 30)
+
+        # Initialize and show the export dialog
+        dialog = ExportClipAndLabelsDialog(video_fps=video_fps)
+        if dialog.exec_() != QtWidgets.QDialog.Accepted:
+            return False  # User canceled
+
+        # Get user input from dialog
+        export_options = dialog.get_results()
+
+        # Prompt user to select output file
+        filename, _ = QtWidgets.QFileDialog.getSaveFileName(
+            None, "Save Clip As...", "", "Video (*.mp4 *.avi)"
+        )
+        if not filename:
+            return False  # User canceled file selection
+
+        # Populate export parameters
+        params["filename"] = filename
+        params["fps"] = export_options["fps"]
+        params["open_when_done"] = export_options["open_when_done"]
+
+        # Access frame range
+        if context.state.get("frame_range"):
+            params["frames"] = range(*context.state["frame_range"])
+        else:
+            params["frames"] = range(context.state["video"].frames)
+
+        return True
+
+
+class ExportClipPkg(AppCommand):
+    @staticmethod
+    def do_action(context: CommandContext, params: dict):
+        """
+        Exports a pruned labels package based on selected frame range.
+
+        Args:
+            context (CommandContext): Contains state information like video and labels.
+            params (dict): Parameters including filename, fps, and open_when_done.
+        """
+        # Extract video and labels from context
+        video = context.state["video"]
+        labels = context.state["labels"]
+
+        # Ensure frame range is set; default to all frames if None
+        frame_range = context.state.get("frame_range", (0, video.frames))
+
+        # Check if clip is selected, raise error if no clip selected
+        if frame_range == (0, video.frames) or frame_range == (0, 1) or frame_range[0] == frame_range[1]:
+            raise ValueError("No valid clip frame range selected! Please select a valid frame range using shift + click in the GUI.")
+
+        # Map frame indices to the actual labeled frame objects
+        frame_to_index = {lf.frame_idx: idx for idx, lf in enumerate(labels.labeled_frames) if lf.video == video}
+        valid_frame_indices = [frame for frame in range(*frame_range) if frame in frame_to_index]
+
+        if not valid_frame_indices:
+            raise ValueError("No valid labeled frames found in the selected frame range.")
+
+        # Extract only the selected frames into a new Labels object
+        pruned_labels = labels.extract(
+            inds=[frame_to_index[frame] for frame in valid_frame_indices],
+            copy=True  # Ensures a deep copy of the extracted labels
+        )
+
+        # Remap frame indices in pruned_labels to start from 0
+        for labeled_frame in pruned_labels.labeled_frames:
+            labeled_frame.frame_idx -= frame_range[0]
```

Comment:
A lot of this `do_action` is the same as the `ExportClipVideo` class... do you think there is a way we  can maybe break out the subparts of `do_action` into class methods and either combine the `ExportClipVideo` and `ExportClipPkg` classes into one class OR create create a base class and subclass to re-use code?